### DISCUSSION TITLE
Linking to a commit in the context of a pull request comment

### DISCUSSION THREAD

**Discussion by jamesgarfield** [community/community#61380] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

In a typical pull request workflow, reviewers ask for changes in comments or discussion. When the implementers respond to those change requests, it's common to respond in the discussion with a link to that commit so the reviewer knows where to find the fix.

The easy way to do this is to get a copy of the commit hash and paste it into the conversation, where Github attempts to helpfully create a link to the commit. However the link that GitHub creates is to the commit _outside the context_ of the Pull Request (i.e. `/$repo/commit/$hash` instead of `$repo/pull/$pr/commits/$hash`)  This means that if a reviewer follows this link and then creates a comment on the code there, this comment does not show up in the context of the commit and the implementer may never see it.

It would be helpful if when making a link from a commit hash in the context of a PR discussion, if the link was created to the commit in the context of the PR instead of the raw commit to help keep discussions centralized.

**Comment by IdaParks** [comment]
I am not an expert but I just want to say that Replace 61380 with the number of the pull request you want to reference. When you post this comment in the discussion, GitHub will automatically convert the number into a clickable link to the pull request. This way, the link will be in the context of the pull request, and any discussions or comments made on that commit will be visible within the pull request in [mcdvoice](https://mcdvoice.me/).

**Reply by jamesgarfield** [reply]
Sorry for the misunderstanding, I'm not trying to link to _just_ a specific PR, I'm trying to create links to a specific commit _within_ the PR when participating in discussions in that same PR.

**Comment by samueltlg** [comment]
Would be useful

**Comment by hk2281** [comment]
Via the web UI

Open your PR and switch to the Files changed tab.
Scroll to the hunk you care about.
Hover over the gutter (the line‑number column) and click the line number at the start of the hunk. That single click will highlight that line and append an anchor like #diff-<file‑hash>R<start> to your URL.
If you want to highlight a whole range, hold Shift and click the ending line number; your URL will look like:
`https://github.com/<owner>/<repo>/pull/<number>/files#diff-<file‑hash>R<start>R<end>`

> ### Выберите тему
> Отзыв о продукте
> 
> ### Тело
> В типичном рабочем процессе pull request рецензенты просят внести изменения в комментариях или в обсуждении. Когда реализаторы отвечают на эти запросы на изменение, обычно в обсуждении они дают ссылку на этот коммит, чтобы рецензент знал, где найти исправление.
> 
> Самый простой способ сделать это — получить копию хэша коммита и вставить ее в беседу, где Github попытается услужливо создать ссылку на коммит. Однако ссылка, которую создает GitHub, ведет к коммиту _вне контекста_ запроса на извлечение (т. е. `/$repo/commit/$hash`вместо `$repo/pull/$pr/commits/$hash`). Это означает, что если рецензент перейдет по этой ссылке, а затем создаст там комментарий к коду, этот комментарий не отобразится в контексте коммита, и реализатор может его никогда не увидеть.
> 
> Было бы полезно, если бы при создании ссылки из хэша коммита в контексте обсуждения PR, ссылка была создана на коммит в контексте PR, а не на сырой коммит, чтобы помочь сохранить централизацию обсуждений.
